tests/unit/modules/state_test.py

```diff
@@ -0,0 +1,993 @@
+                Mock compile_low_chunks method
+            '''
+            return True
+
+        def render_highstate(self, data):
+            '''
+                Mock render_highstate method
+            '''
+            data = data
+            if self.flag:
+                return ["a", "b"], True
+            else:
+                return ["a", "b"], False
+
+        @staticmethod
+        def call_highstate(exclude, cache, cache_name, force=None,
+                           whitelist=None):
+            '''
+                Mock call_highstate method
+            '''
+            exclude = exclude
+            cache = cache
+            cache_name = cache_name
+            force = force
+            whitelist = whitelist
+            return True
+
+
+class MockSerial(object):
+    '''
+        Mock Class
+    '''
+    def __init__(self):
+        pass
+
+    class Serial(object):
+        '''
+            Mock Serial class
+        '''
+        def __init__(self, data):
+            data = data
+
+        @staticmethod
+        def load(data):
+            '''
+                Mock load method
+            '''
+            data = data
+            return {"A": "B"}
+
+        @staticmethod
+        def dump(data, data1):
+            '''
+                Mock dump method
+            '''
+            data = data
+            data1 = data1
+            return True
+
+
+class MockTempFile(object):
+    '''
+        Mock tempfile class
+    '''
+
+    @staticmethod
+    def mkdtemp():
+        '''
+            mock mkdtemp method
+        '''
+        return "/tmp"
+
+
+class MockTarFile(object):
+    '''
+        Mock tarfile class
+    '''
+    path = "/tmp"
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def open(data, data1):
+        '''
+            Mock open method
+        '''
+        data = data
+        data1 = data1
+        return MockTarFile
+
+    @staticmethod
+    def getmembers():
+        '''
+            Mock getmembers method
+        '''
+        return [MockTarFile]
+
+    @staticmethod
+    def extractall(data):
+        '''
+            Mock extractall method
+        '''
+        data = data
+        return True
+
+    @staticmethod
+    def close():
+        '''
+            Mock close method
+        '''
+        return True
+
+
+class MockJson(object):
+    '''
+        Mock json class
+    '''
+    flag = None
+
+    def __init__(self):
+        pass
+
+    def load(self, data, object_hook=None):
+        '''
+            Mock load method
+        '''
+        data = data
+        object_hook = object_hook
+        if self.flag:
+            return [True]
+        else:
+            return [{"test": ""}]
+
+
+class MockShutil(object):
+    '''
+        Mock shutil class
+    '''
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def rmtree(data):
+        '''
+            Mock rmtree
+        '''
+        data = data
+        return True
+
+
+state.shutil = MockShutil
```

Comment:
These mocks should be `@patch`ed on the test case class or `@patch`ed on `setUp` and `tearDown`.

Like this it overrides salts modules globably and other test cases need the real modules, not the patched ones.

Unittest2 does not clear `sys.modules` for each test module, that's why this is problematic.